addons/account/wizard/account_payment_register.py

```diff
@@ -409,6 +409,7 @@ def _compute_group_payment(self):
                 wizard.group_payment = len(batches[0]['lines'].move_id) == 1
             else:
                 wizard.group_payment = False
+            wizard.group_payment = self._context.get('force_group_payment', wizard.group_payment)
```

Comment:
If `force_group_payment` is not None, there is no need to loop on all records and perform some computations (such as get batches). You could just do `self.group_payment = self._context.get('force_group_payment')` in the beginning, if not you can loop